src/extension/debugConfigurationProvider.ts

```diff
@@ -0,0 +1,115 @@
+            "outDir": "${workspaceRoot}/.vscode/.react",
+        },
+        "Attach to packager": {
+            "name": "Attach to packager",
+            "program": "${workspaceRoot}/.vscode/launchReactNative.js",
+            "type": "reactnative",
+            "request": "attach",
+            "sourceMaps": true,
+            "outDir": "${workspaceRoot}/.vscode/.react",
+        },
+        "Debug in Exponent": {
+            "name": "Debug in Exponent",
+            "program": "${workspaceRoot}/.vscode/launchReactNative.js",
+            "type": "reactnative",
+            "request": "launch",
+            "platform": "exponent",
+            "sourceMaps": true,
+            "outDir": "${workspaceRoot}/.vscode/.react",
+        }};
+
+    private pickConfig: ReadonlyArray<vscode.QuickPickItem> = [
+        {
+            label: "Debug Android",
+            description: localize("DebugAndroidConfigDesc", "Run and debug Android application"),
+            picked: true,
+        },
+        {
+            label: "Debug iOS",
+            description: localize("DebugiOSConfigDesc", "Run and debug iOS application"),
+        },
+        {
+            label: "Attach to packager",
+            description: localize("AttachToPackagerConfigDesc", "Attach to already working application packager"),
+        },
+        {
+            label: "Debug in Exponent",
+            description: localize("DebugExpoConfigDesc", "Debug Expo application or React Native application in Expo"),
+        },
+    ];
+
+    public async provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): Promise<vscode.DebugConfiguration[]> {
+        const configPicker = this.prepareDebugConfigPicker();
+        configPicker.show();
+        const disposables: vscode.Disposable[] = [];
+        return new Promise<vscode.DebugConfiguration[]>((resolve) => {
+            const pickHandler = () => {
+                let chosenConfigsEvent = TelemetryHelper.createTelemetryEvent("ChosenDebugConfigurations");
+                let selected: string[] = [];
+                configPicker.selectedItems.forEach((element) => {
+                    selected.push(element.label);
+                });
+                chosenConfigsEvent.properties["SelectedItems"] = selected;
+                Telemetry.send(chosenConfigsEvent);
+                const launchConfig = this.gatherDebugScenarios(selected);
+                disposables.forEach(d => d.dispose());
+                configPicker.dispose();
+                resolve(launchConfig);
+            };
+
+            disposables.push(
+            configPicker.onDidAccept(pickHandler),
+            configPicker.onDidHide(pickHandler)
+            );
+
+        });
+    }
+
+    private gatherDebugScenarios(selectedItems: string[]): vscode.DebugConfiguration[] {
+        let launchConfig: vscode.DebugConfiguration[] = [];
+        const pickedConfigs = selectedItems;
```

Comment:
Is it necessary to have additional `pickedConfigs` variable here?

Reply:
Thanks for pointing. Fixed